src/main/java/org/cryptomator/ui/traymenu/AppindicatorTrayMenuController.java

```diff
@@ -0,0 +1,106 @@
+package org.cryptomator.ui.traymenu;
+
+import org.apache.commons.lang3.SystemUtils;
+import org.cryptomator.integrations.common.CheckAvailability;
+import org.cryptomator.integrations.tray.ActionItem;
+import org.cryptomator.integrations.tray.SeparatorItem;
+import org.cryptomator.integrations.tray.SubMenuItem;
+import org.cryptomator.integrations.tray.TrayMenuController;
+import org.cryptomator.integrations.tray.TrayMenuException;
+import org.cryptomator.integrations.tray.TrayMenuItem;
+import org.purejava.linux.MemoryAllocator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.awt.SystemTray;
+import java.io.File;
+import java.lang.foreign.MemoryAddress;
+import java.lang.foreign.MemorySession;
+import java.net.URISyntaxException;
+import java.nio.file.Paths;
+import java.util.List;
+
+import static org.purejava.linux.app_indicator_h.*;
+
+@CheckAvailability
+public class AppindicatorTrayMenuController implements TrayMenuController {
+
+	private static final Logger LOG = LoggerFactory.getLogger(AppindicatorTrayMenuController.class);
+
+	private final MemorySession session = MemorySession.openShared();
+	private MemoryAddress indicator;
+	private MemoryAddress menu = gtk_menu_new();
+
+	@CheckAvailability
+	public static boolean isAvailable() {
+		return SystemUtils.IS_OS_LINUX && MemoryAllocator.isLoadedNativeLib() && SystemTray.isSupported();
```

Comment:
`SystemTray.isSupported()` should not be needed, since this only determines whether the AWT system tray is supported, which might not be the case while libappindicator is still available.